Create GroupMembership block condition plugin

andyg5000 opened the pull request:
Create the ability to show/hide Drupal blocks based on the users membership in the group loaded from OG context.

andyg5000 committed:
#350: Created GroupMembership block condition plugin

andyg5000 committed:
#350: Added GroupMembershipConditionTest

andyg5000 committed:
#350: PHPCS fixes for GroupMembershipConditionTest

andyg5000 committed:
#350: PHPCS Missing short description in doc comment

andyg5000 committed:
submitConfigurationForm() override caused configuration to always be set. Let the parent method handle saving config values

MPParsley commented on src/Plugin/Condition/GroupMembership.php:
```diff
@@ -0,0 +1,124 @@
+    }
+
+    // The membership status is not active.
+    if ($membership->getState() != OgMembership::STATE_ACTIVE) {
```
```suggestion
    if ($membership->getState() !== OgMembership::STATE_ACTIVE) {
```

MPParsley commented on src/Plugin/Condition/GroupMembership.php:
```diff
@@ -0,0 +1,124 @@
+ *   }
+ * )
+ */
+class GroupMembership extends ConditionPluginBase {
```
Perhaps we can rename this to OgGroupMembership.

MPParsley committed:
Merge branch '8.x-1.x' into 350-og-membership-condition

MPParsley committed:
Use type specific test

MPParsley commented:
Should we group the group conditions in one plugin/tab? See #265

![block-visibility-plugin](https://cloud.githubusercontent.com/assets/442924/25944536/f3b6ac82-3643-11e7-93d7-3fa7cf59bf5d.png)

MPParsley commented on src/Plugin/Condition/GroupMembership.php:
```diff
@@ -0,0 +1,124 @@
+ * @Condition(
+ *   id = "og_group_membership",
+ *   label = @Translation("Group Membership"),
+ *   context = {
```
Providing context definitions via the "context" key is deprecated in Drupal 8.7.x and will be removed before Drupal 9.0.0. Use the "context_definitions" key instead.

```suggestion
 *   context_definitions = {
```

MPParsley committed:
Deprecate context definintion with context_definitions

MPParsley commented on tests/src/Kernel/GroupMembershipConditionTest.php:
```diff
@@ -0,0 +1,180 @@
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('og_membership');
+    $this->installSchema('system', ['queue', 'sequences']);
```
Special handling of system module schemas in \Drupal\KernelTests\KernelTestBase::installSchema has been deprecated in Drupal 8.7.x, remove any calls to this method that use invalid schema names. See https://www.drupal.org/node/3003360.

```suggestion
    $this->installSchema('system', ['sequences']);
```

MPParsley committed:
Remove call to invalid schema queue

Special handling of system module schemas in \Drupal\KernelTests\KernelTestBase::installSchema has been deprecated in Drupal 8.7.x, remove any calls to this method that use invalid schema names. See https://www.drupal.org/node/3003360.

MPParsley commented on src/Plugin/Condition/GroupMembership.php:
```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\og\Plugin\Condition;
```
```suggestion
declare(strict_types = 1);

declare(strict_types = 1);
namespace Drupal\og\Plugin\Condition;
```

MPParsley commented on tests/src/Kernel/GroupMembershipConditionTest.php:
```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Drupal\Tests\og\Kernel;
```
```suggestion
declare(strict_types = 1);

namespace Drupal\Tests\og\Kernel;
```

MPParsley committed:
Add strict types

MPParsley committed:
Merge branch '8.x-1.x' into 350-og-membership-condition

MPParsley committed:
Merge branch '8.x-1.x' into 350-og-membership-condition

claudiu-cristea commented:
Since https://www.drupal.org/project/og/issues/3467694 was merged into the `8.x-1.x` branch, the development has been moved to https://www.drupal.org/project/og. If this change is still actual, please open an issue at https://www.drupal.org/project/issues/og and provide there a _merge request_ similar with this pull request.

Thank you for contributing to Organic groups module.
